### PR TITLE
New adaptive mesh refinement criteria based on the local gradient magnitude and curvature

### PR DESCRIPTION
**Pull request by gdeskos**

### CODE REVIEW

File: amr-wind/utilities/tagging/CMakeLists.txt [hunk 1]
```diff
@@ -3,12 +3,17 @@ target_sources(${amr_wind_lib_name} PRIVATE
...
+  VorticityMagRefinement.cpp
   OversetRefinement.cpp
   GeometryRefinement.cpp
 
   # Geometry based refinement options
   BoxRefiner.cpp
   CylinderRefiner.cpp
+
+  # Multiphase based refinement option
+
```
Comment (sayerhs) [inline amr-wind/utilities/tagging/CMakeLists.txt:18]:
```suggestion
```

File: amr-wind/utilities/tagging/CurvatureRefinement.H [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *  \ingroup amr_utils
+ *
+ *  ```
+ *  tagging.labels = t1
+ *  tagging/t1.type = FieldRefinement
+ *  tagging/t1.field_name = density
+ *  # Tagging based on field value
+ *  # tagging/t1.field_error = 10.0 10.0
+
+ */
```
Comment (sayerhs) [inline amr-wind/utilities/tagging/CurvatureRefinement.H:19]: Update comment for curvature.

File: amr-wind/utilities/tagging/CurvatureRefinement.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    amrex::ParmParse pp(key);
+    std::string fname;
+    pp.query("field_name", fname);
+
+    const auto& repo = m_sim.repo();
+    if (!repo.field_exists(fname)) {
+        amrex::Abort("CurvatureRefinement: Cannot find field = " + fname);
+    }
+    m_field = &(m_sim.repo().get_field(fname));
+
```
Comment (sayerhs) [inline amr-wind/utilities/tagging/CurvatureRefinement.cpp:26]: Is this needed? Or is the field always going to be `curvature`?

File: amr-wind/utilities/tagging/GradientMagRefinement.H [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *  \ingroup amr_utils
+ *
+ *  ```
+ *  tagging.labels = t1
+ *  tagging/t1.type = FieldRefinement
+ *  tagging/t1.field_name = density
+ *  # Tagging based on field value
+ *  # tagging/t1.field_error = 10.0 10.0
+
+ */
```
Comment (sayerhs) [inline amr-wind/utilities/tagging/GradientMagRefinement.H:19]: Update comment for magnitude-of-gradient tagging

File: amr-wind/utilities/tagging/CurvatureRefinement.cpp (hunk 3 above)
Comment (gdeskos) [reply amr-wind/utilities/tagging/CurvatureRefinement.cpp:26]: Well you need to define on which field curvature needs to be computed. I use density for vof but you may use any field you want

**Pull request merged**